src/main/java/jenkins/plugins/slack/webhook/WebhookEndpoint.java

```diff
@@ -0,0 +1,245 @@
+package jenkins.plugins.slack.webhook;
+
+
+import jenkins.model.Jenkins;
+import jenkins.model.GlobalConfiguration;
+
+import hudson.Extension;
+
+import hudson.model.Build;
+import hudson.model.Result;
+import hudson.model.Project;
+import hudson.model.AbstractBuild;
+import hudson.model.AbstractProject;
+import hudson.model.UnprotectedRootAction;
+
+import hudson.security.ACL;
+
+import javax.servlet.ServletException;
+
+import java.io.IOException;
+
+import java.util.List;
+import java.util.ArrayList;
+
+import java.util.logging.Logger;
+
+import org.kohsuke.stapler.HttpResponse;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.StaplerResponse;
+
+import org.kohsuke.stapler.interceptor.RequirePOST;
+
+import jenkins.plugins.slack.webhook.model.JsonResponse;
+import jenkins.plugins.slack.webhook.model.SlackPostData;
+import jenkins.plugins.slack.webhook.model.SlackTextMessage;
+import jenkins.plugins.slack.webhook.model.SlackWebhookCause;
+
+import jenkins.plugins.slack.webhook.exception.CommandRouterException;
+import jenkins.plugins.slack.webhook.exception.RouteNotFoundException;
+
+
+
+
+@Extension
+public class WebhookEndpoint implements UnprotectedRootAction {
+
+    private String slackUser;
+
+    private GlobalConfig globalConfig;
+
+    private static final Logger LOGGER =
+        Logger.getLogger(WebhookEndpoint.class.getName());
+
+    public WebhookEndpoint() {
+        globalConfig = GlobalConfiguration.all().get(GlobalConfig.class);
+    }
+
+    @Override
```

Comment:
The pom file specifies that the compiler should target JDK 1.5, which actually means that these @Overrides shouldn't be allowed because they're implementations of an interface rather than overrides of a parent class, which was only allowed from 1.6 on. I definitely prefer to have the annotation, and it works for building because our more modern compilers aren't strictly enforcing it, but my IDE doesn't like them.

However, why are we stuck using 1.5? I followed the path of where it comes from:

    org.jenkins-ci.plugins.slack pom.xml
        --> org.jenkins-ci.plugins.plugin 1.567
            -> org.jenkins-ci.jenkins 1.33

org.jenkins-ci.jenkins 1.33 was last updated in 2013, so maybe we can move to the latest org.jenkins-ci.plugins.plugin. Even that will only give us 1.6, which isn't a supported version of Java anymore.

Not sure this has anything to do with this review, but @samrocketman do you know if there are any plans to move Jenkins in general to a newer version of Java? Otherwise, maybe the next thing I can work on is updating these versions to the latest.